crates/llm-chain-llama/src/executor.rs

```diff
@@ -147,47 +157,36 @@ impl Executor {
                 {
                     break;
                 }
-                if tok == tokenized_stop_prompt[stop_sequence_i] {
+                // If the prompt is in the form of a question then next
+                // predicted tok will be a new line.
+                if tok == tokenized_stop_prompt[stop_sequence_i] && n_samples > 2 {
```

Comment:
I don't fully understand the n_samples > 2. If you're trying to handle another case. Could we break it up into another if case

Reply:
I think I got it. But what if you actually expect a no answer?

Reply:
Ah good point, let me think about that and revisit this. 

@Juzov Thanks for your review and comments, I appreciate the feedback!

Reply:
I've taken another look at this in https://github.com/sobelio/llm-chain/pull/244/commits/d73826bd21fc761266222222382f639f3cf2f13e. I've not been able to get the llm to respond with nothing yet, but I've extracted this into a separate check like you suggested and I think this should take care of the case when an input prompt is in the form of a question.